src/mlpack/core/data/string_encoding.hpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file string_encoding.hpp
+ * @author Jeffin Sam
+ *
+ * Definition of string encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_STRING_ENCODING_HPP
+#define MLPACK_CORE_DATA_STRING_ENCODING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "mlpack/core/boost_backport/boost_backport_string_view.hpp"
+#include <mlpack/core/data/encoding_policies/dictionary_encoding.hpp>
+#include <mlpack/core/data/encoding_policies/policy_traits.hpp>
+#include <utility>
+
+using MapType =   std::unordered_map<boost::string_view, size_t,
+    boost::hash<boost::string_view>>;
+
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple String Enocding class
+ *
+ * The encoding here simply assigns a word (or a character) to a numeric
+ * index and treat the dataset as categorical.The assignement of numeric index
+ * is based on the EncodingPolicy.
+ */
+class StringEncoding
+{
+ public:
+  /**
+  * A function to create mapping from a given corpus.
+  * 
+  * @param input Corpus of text to encode.
+  * @param tokenizer A function that accepts a boost::string_view as
+  *                  an argument and returns a token.
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be a boost::string_view, a token.
+  *
+  * Definition of function should be of type
+  * boost::string_view fn(boost::string_view& str)
+  *
+  */
+  template<typename TokenizerType>
+  void CreateMap(std::string& input, TokenizerType tokenizer);
+  /**
+  * Default Constructor
+  */
+  StringEncoding() {}
+  /**
+  * Copy Constructor.
+  */
+  StringEncoding(const StringEncoding& oldObject);
+  /*
+  * Move Constructor.
+  */
+  StringEncoding(StringEncoding&& oldObject) = default;
+  /*
+  * Move Assignment Operator.
+  */  
+  StringEncoding& operator= (StringEncoding&& oldObject) = default;
+  /*
+  * Assignment Operator.
+  */
+  StringEncoding& operator= (const StringEncoding& oldObject);
+  /**
+  * A function to reset the mapping that is clear all the encodings
```

Comment:
There is a dot missing at the end of the sentence.

Reply:
Rectified